src/live_restore/live_restore_fs.c

```diff
@@ -288,6 +288,8 @@ __live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_sessi
     uint32_t count_dest = 0, count_src = 0;
```

Comment:
There's a few bug fixes in here but the diff makes it unclear what they're doing. I'll leave comments for each to explain what's changed.